python_modules/dagster/dagster_tests/general_tests/test_serdes.py

```diff
@@ -193,6 +193,39 @@ def __new__(cls, foo, bar):
     assert not hasattr(deserialized, "baz")
 
 
+def test_forward_compat():
+    old_map = WhitelistMap.create()
+
+    @_whitelist_for_serdes(whitelist_map=old_map)
+    class Quux(namedtuple("_Quux", "bar baz")):
+        def __new__(cls, bar, baz):
+            return super().__new__(cls, bar, baz)
+
+    # new version has a new field with a new type
+    new_map = WhitelistMap.create()
+
+    # pylint: disable=function-redefined
+    @_whitelist_for_serdes(whitelist_map=new_map)
+    class Quux(namedtuple("_Quux", "foo bar baz")):
+        def __new__(cls, foo, bar, baz):
+            return super().__new__(cls, foo, bar, baz)
+
+    @_whitelist_for_serdes(whitelist_map=new_map)
+    class Foo(namedtuple("_Foo", "wow")):
+        def __new__(cls, wow):
+            return super().__new__(cls, wow)
+
+    new_quux = Quux(foo=Foo("wow"), bar="bar", baz="baz")
+
+    # write from new
+    serialized = _serialize_dagster_namedtuple(new_quux, whitelist_map=new_map)
+
+    # read from old, foo ignored
+    deserialized = _deserialize_json(serialized, whitelist_map=old_map)
+    assert deserialized.bar == "bar"
+    assert deserialized.baz == "baz"
+
```

Comment:
add a test case here with an attribute rename (to catch the `PipelineRun` case)?

Reply:
added a test case